Create initial project structure and first admin-related endpoints

accordeiro opened the pull request:
This PR:
- Closes #1 
- Closes #7

It aims to create an initial structure for this TypeScript project, providing:
- Database connection (sqlite)
- User Login / Sign up
- Authentication
- Authorization
- API endpoints for CRUDing users
- API endpoint testing
- Documentation on how to run the project

ire-and-curses commented on src/controllers/AuthController.ts:
```diff
@@ -0,0 +1,59 @@
+      return;
+    }
+
+    if (!user.checkIfUnencryptedPasswordIsValid(password)) {
```
could consider shortening name to `user.unecryptedPasswordIsValid`. I don't know the TS convention, but this is the best practice in e.g. Perl (see p.40 here ftp://cvs.happy.kiev.ua/pub/doc/Damian%20Conway%20-%20Perl%20Best%20Practices%20-%202005.pdf)

accordeiro replied:
Agreed – I've simplified the logic / updated the naming. Thanks!

ire-and-curses commented on src/controllers/UserController.ts:
```diff
@@ -0,0 +1,100 @@
+    const user = new User();
+    user.username = username;
+    user.password = password;
+    user.role = role;
```
Can these assignments be done as part of the `new User()` call?

accordeiro replied:
Definitely! Just pushed a commit to address this.

ire-and-curses commented on src/index.ts:
```diff
@@ -0,0 +1,14 @@
+  .then(async connection => {
+    app.listen(settings.PORT, () => {
+      console.log(
+        `🚀 Anchor server has started on http://localhost:${settings.PORT}`,
```
Nice touch with the rocket :rocket:

ire-and-curses commented on src/migration/1560357776337-CreateAdminUser.ts:
```diff
@@ -0,0 +1,20 @@
+config();
+const adminPassword = process.env.INITIAL_ADMIN_PASSWORD || "admin";
+
+export class CreateAdminUser1560357776337 implements MigrationInterface {
```
What does the strangely specific number refer to? Maybe add a comment

accordeiro replied:
This was automatically generated by TypeORM – I've added a comment in the file to make that clearer.

accordeiro commented on README.md:
```diff
@@ -1 +1,19 @@
+
+This project is a WIP example implementation of a Stellar anchor server.
+
+Its goal is to provide a community example implementation of SEP 6 (and related SEPs 9, 10 and 12) to make it easier for anchors to integrate with the Stellar network, and enable wallets to seamlessly integrate with said anchor.
```
Done! :)

accordeiro commented on README.md:
```diff
@@ -1 +1,19 @@
+
+## Running the project locally
+
+1. Install npm
```
Done – I've added the main tools' versions before the steps :)

accordeiro commented:
Yes, I was definitely being light on types – I've changed the `tsconfig.json` to disallow implicit `any`s and implicit return `any`s, and updated the code to reflect that. Thanks for the review!
